modules/reactor/src/meshgenerators/ControlDrumMeshGenerator.C

```diff
@@ -0,0 +1,517 @@
+                                                               360. / (Real)num_sectors))
+    paramError("azimuthal_node_tolerance",
+               "Azimuthal node tolerance should be smaller than half the azimuthal interval size "
+               "as defined by 'num_azimuthal_sectors'");
+  if (_has_pad_region && MooseUtils::absoluteFuzzyGreaterEqual(2. * azimuthal_node_tolerance,
+                                                               _pad_end_angle - _pad_start_angle))
+    paramError("azimuthal_node_tolerance",
+               "Azimuthal node tolerance should be smaller than half the difference of the pad "
+               "angle range");
+
+  // Check region IDs have correct size
+  const unsigned int n_radial_regions = _has_pad_region ? 4 : 3;
+  if (isParamValid("region_ids"))
+  {
+    unsigned int n_axial_levels =
+        (_mesh_dimensions == 3)
+            ? getReactorParam<std::vector<unsigned int>>(RGMB::axial_mesh_intervals).size()
+            : 1;
+    if (_region_ids.size() != n_axial_levels)
+      mooseError("The size of region IDs must be equal to the number of axial levels as defined in "
+                 "the ReactorMeshParams object");
+    if (_region_ids[0].size() != n_radial_regions)
+    {
+      std::string err_msg =
+          "'region_ids' parameter does not have the correct number of elements per axial zone. ";
+      err_msg += _has_pad_region
+                     ? "For control drums with a pad region, 4 radial IDs need to be provided per "
+                       "axial zone (drum inner, drum pad, drum ex-pad, and drum outer)"
+                     : "For control drums with no pad region, 3 radial IDs need to be provided per "
+                       "axial zone (drum inner, drum, and drum outer)";
+      paramError("region_ids", err_msg);
+    }
+  }
+  else
+    mooseError(
+        "Region IDs must be assigned for ControlDrumMeshGenerator using parameter 'region_ids'");
+
+  // Check block names have the correct size
+  if (isParamValid("block_names"))
+  {
+    if (getReactorParam<bool>(RGMB::region_id_as_block_name))
+      paramError("block_names",
+                 "If ReactorMeshParams/region_id_as_block_name is set, block_names should not be "
+                 "specified in ControlDrumMeshGenerator");
+    _has_block_names = true;
+    _block_names = getParam<std::vector<std::vector<std::string>>>("block_names");
+    if (_region_ids.size() != _block_names.size())
+      mooseError("The size of block_names must match the size of region_ids");
+    for (const auto i : index_range(_region_ids))
+      if (_region_ids[i].size() != _block_names[i].size())
+        mooseError("The size of block_names must match the size of region_ids");
+  }
+  else
+    _has_block_names = false;
+
+  // Check extrusion parameters
+  if (_extrude && _mesh_dimensions != 3)
+    paramError("extrude",
+               "This is a 2 dimensional mesh, you cannot extrude it. Check your ReactorMeshParams "
```

Comment:
```suggestion
               "Reactor mesh has been set to two dimensions, you cannot extrude it. Check your ReactorMeshParams "
```

Reply:
Updated the error message to make it more informative